### ISSUE TITLE
scipy.stats.multinomial.pmf is numerically incorrect.

### ISSUE THREAD

**Issue by helmutsimon** [scipy/scipy#8811]
scipy.stats.multinomial.pmf appears to be numerically incorrect as per examples in code below.

```
import numpy as np
from scipy.stats import multinomial
from math import log, factorial


def multinomial_pmf(counts, probs):
    x = np.prod(probs ** counts)
    y = np.prod([factorial(i) for i in counts])
    z = factorial(np.sum(counts))
    return x * z / y


def multinomial_pmf_log(counts, probs):
    xx = np.log(probs) * counts
    counts = np.array(counts)
    xx[np.where(counts == 0.)] = 0.                # otherwise x will contain NaN even if count = 0
    x = np.sum(xx)
    y = np.sum([log(factorial(i)) for i in counts])
    z = np.sum(np.log(np.arange(1, sum(counts) + 1)))
    mult_prob = np.exp(x + z - y)
    return mult_prob


counts = np.array([0,3,0,0,3,1])
p1 = np.array([9.495e-01, 3.261e-02, 1.123e-02, 4.170e-03, 2.165e-03, 3.633e-04])
p2 = np.array([9.289e-01, 6.639e-02, 0.0, 2.921e-03, 1.238e-03, 5.623e-04])
print('\n', multinomial.pmf(counts, np.sum(counts), p1))
print(multinomial_pmf(counts, p1))
print(multinomial_pmf_log(counts, p1))
print('\n', multinomial.pmf(counts, np.sum(counts), p2))
print(multinomial_pmf(counts, p2))
print(multinomial_pmf_log(counts, p2))
```

rsb0001148:bayescoalescentest helmutsimon$ python -c "import scipy; print(scipy.__version__)"
1.1.0

**Comment by WarrenWeckesser** [comment]
@helmutsimon Thanks for bug report.

The second problem, where `pmf` returns `nan`, has been fixed in version 1.1.0 of scipy (see the original bug report at https://github.com/scipy/scipy/issues/8235 and the fix at https://github.com/scipy/scipy/pull/8236).

A problem with your first example is that the probabilities do not sum to 1:

```
In [3]: p1 = np.array([9.495e-01, 3.261e-02, 1.123e-02, 4.170e-03, 2.165e-03, 3.633e-04])

In [4]: sum(p1)
Out[4]: 1.0000382999999999
```
If I normalize `p1` by dividing the probabilities by `p1.sum()`, the numerical discrepancy goes away.  Here's the output of a modified version of your script (given below), using the development version of scipy:

```
1.78938214665e-14
1.78938214665e-14
1.78938214665e-14


4.37050585803e-14
4.37050585803e-14
/Users/warren/tmps/multinomial_bug_report.py:14: RuntimeWarning: divide by zero encountered in log
  
/Users/warren/tmps/multinomial_bug_report.py:14: RuntimeWarning: invalid value encountered in multiply
  
4.37050585803e-14
```

Here's the modified script:

```
import numpy as np
from scipy.stats import multinomial
from math import log, factorial


def multinomial_pmf(counts, probs):
    x = np.prod(probs ** counts)
    y = np.prod([factorial(i) for i in counts])
    z = factorial(np.sum(counts))
    return x * z / y


def multinomial_pmf_log(counts, probs):
    xx = np.log(probs) * counts
    counts = np.array(counts)
    xx[np.where(counts == 0.)] = 0.                # otherwise x will contain NaN even if count = 0
    x = np.sum(xx)
    y = np.sum([log(factorial(i)) for i in counts])
    z = np.sum(np.log(np.arange(1, sum(counts) + 1)))
    mult_prob = np.exp(x + z - y)
    return mult_prob


counts = np.array([0,3,0,0,3,1])
p1 = np.array([9.495e-01, 3.261e-02, 1.123e-02, 4.170e-03, 2.165e-03, 3.633e-04])
p1 /= p1.sum()
p2 = np.array([9.289e-01, 6.639e-02, 0.0, 2.921e-03, 1.238e-03, 5.623e-04])
p2 /= p2.sum()
print('\n')
print(multinomial.pmf(counts, np.sum(counts), p1))
print(multinomial_pmf(counts, p1))
print(multinomial_pmf_log(counts, p1))
print('\n')
print(multinomial.pmf(counts, np.sum(counts), p2))
print(multinomial_pmf(counts, p2))
print(multinomial_pmf_log(counts, p2))
```

**Comment by helmutsimon** [comment]
You are correct. Sorry for wasting your time.
